docs/release/qa.md

```diff
@@ -46,28 +46,43 @@ For both:
     ```sh
     docker image pull mathesar/mathesar-caddy:<version_number>
     docker image tag mathesar/mathesar-caddy:<version_number> mathesar/mathesar-caddy:latest
-    
+
     docker image pull mathesar/mathesar:<version_number>
     docker image tag mathesar/mathesar:<version_number> mathesar/mathesar:latest
     ```
 
+### Understanding Database connections
+
+#### Database types
+
+- **Mathesar Database** - The Django database used by Mathesar
+- **Data database** - The user database(s) being explored in Mathesar's UI
+
+#### Connection types
+
+- **Local** - On the host machine
+- **External** - Outside of the host machine
+- **Integrated** - Inside of Mathesar's docker image
+
 ### Tasks
 
 - [ ] Test installing the release from scratch by following our documentation
 - [ ] Test all different installation methods
-    - [ ] Docker compose
-        - [ ] All defaults, local
-        - [ ] Exposed on domain, DB managed by Mathesar
-        - [ ] Exposed on domain, DB preexisting for users, managed DB for Django
-        - [ ] Exposed on domain, DB preexisting for all data
-    - [ ] Docker integrated image
-        - [ ] All defaults, local
-        - [ ] Exposed on domain, DB managed by Mathesar
-        - [ ] Exposed on domain, DB preexisting for users, managed DB for Django
-        - [ ] Exposed on domain, DB preexisting for all data
-    - [ ] Build from scratch
-        - [ ] All defaults, local
-        - [ ] Exposed on domain, DB preexisting for all data
+    - [ ] [Docker compose](https://docs.mathesar.org/latest/administration/install-via-docker-compose/)
+        - [ ] Local using all default settings
+        - [ ] Exposed on domain with different database configurations:
+          - [ ] Integrated Mathesar, Integrated Data
```

Comment:
```suggestion
          - [ ] Local docker pg Mathesar, Local docker pg Data
          - [ ] Local docker pg Mathesar, External Data
```
I'm not sure what's the best way to describe what needs to be tested here is, but basically, for the docker compose method, we want to test mathesar against the `mathesar_db` container that we provide in [our docker-compose config](https://github.com/mathesar-foundation/mathesar/blob/e29a63c7654d3fe66c986023fcaa59786db73495/docker-compose.yml#L185-L206).

Reply:
I know what you mean. There's so many different options here that I forgot that we don't want to use the postgres "baked in" to the mathesar container, we want to use the postgres defined in the docker compose. 

I'll further revise this, thank you for the patience as I learn every layer here one at a time 😅

Reply:
Unsolicited opinion: Would recommend just merging Anish's change and updating terminology later again if needed, this looks like a major improvement that is only blocked by this terminology thing.